src/Compilers/CSharp/Portable/Emitter/Model/PEModuleBuilder.cs

```diff
@@ -1737,6 +1737,23 @@ internal virtual SynthesizedAttributeData SynthesizeScopedRefAttribute(WellKnown
             return Compilation.TrySynthesizeAttribute(member, isOptionalUse: true);
         }
 
+        internal SynthesizedAttributeData SynthesizeRequiresLocationAttribute(ParameterSymbol symbol)
+        {
+            if ((object)Compilation.SourceModule != symbol.ContainingModule)
+            {
+                // For symbols that are not defined in the same compilation (like NoPia), don't synthesize this attribute.
```

Comment:
Do we have a test for this branch?

Reply:
To test this, I think I need to have the next PR in pipeline merged first so I can use a `ref readonly` parameter from another assembly (in this PR, it's not possible to call/override `ref readonly` methods yet).

Reply:
Then I would leave a PROTOTYPE comment here or a `Debug.Assert(false)` to indicate this is not yet covered. Thanks